pepper-apis/src/main/java/org/broadinstitute/ddp/elastic/ElasticSearchQueryExecutor.java

```diff
@@ -0,0 +1,71 @@
+package org.broadinstitute.ddp.elastic;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.index.query.QueryBuilder;
+
+
+/**
+ * Executes query to ElasticSearch.
+ */
+public class ElasticSearchQueryExecutor<T> {
+
+    protected final RestHighLevelClient esClient;
+
+    protected String query;
+    protected String esIndex;
+    protected String[] fetchSource;
```

Comment:
You might consider adding a comment here to tell people to look at ESSearchUtil for the list of options.  Alternately, you might consider creating an enum type for the different sources so that you could have something like `protected Source[] fetchSources;` here, which would naturally lead people to look at the Source type.  Just an option to consider.

Reply:
Yes, I added the comment with code example to the class-level comment. And in the comment I replaced reference to ESSearchUtil to string constants (in order to be clear what these constants mean). 
And I created enum class `IndexSource` where defined all sources and implemented a method to get list of sources for each of used indices.
Hopefully the API now became a bit more clear.

Reply:
Nice!